Add a hook documentation (#1861)
PrestaShop/docs: contribute/contribute-pull-requests/add_hook.md

> # How To add a new Hook

Reviewer: ```suggestion
# How to add a new Hook
```

> Adding a new hook is quite a simple process. It can be done in only tree steps.

Reviewer 2: ```suggestion
Adding a new hook is a process that can be done in only three steps.
```

We should refrain from using "simple", "easy", etc., in the documentation. It might not be that simple for some :)

## 1) Add a call to Hook::exec()

> As you can see on this [PR](https://github.com/PrestaShop/PrestaShop/pull/34431/files) Adding a simple Hook::exex() statement is enought to add a new hook. In order for it to be complete, there is a couple more steps to do.

Reviewer: ```suggestion
As you can see on this [PR](https://github.com/PrestaShop/PrestaShop/pull/34431/files), adding a simple Hook::exex() statement is enough to add a new hook. In order for it to be complete, there is a couple more steps to do.
```

Reviewer 3: `exec`

Reviewer 2: I would be more precise here. In the shown example, you don't write about "adding a hook" but adding code that executes the hook in this place. That doesn't mean the hook is fully added to the system if you add this code.

Author: This is the only way I was told how to do it maybe you can enlight me ?

Reviewer 4: I think @kpodemski means that adding the `Hook::exec()` (part 1) will trigger the hook, but if you dont do the parts 2 and parts 3 it will work only partially

> ## 2) Modify the hook.xlm file

Reviewer: ```suggestion
## 2) Modify the hook.xml file
```

In order to properly register the hook you need to add it to the hook.xml file as seen in this [PR](https://github.com/PrestaShop/PrestaShop/pull/34431/files)

## 3) Add the new hook in the Autoupgrade fixture

> Finally, to allow your new hook to be available during an update, you need to add it to the current version SQL script of the Autoupgrade module. As it can be seen in this [PR](https://github.com/PrestaShop/autoupgrade/pull/672/files)

Reviewer: ```suggestion
Finally, to allow your new hook to be available during an update, you need to add it to the current version SQL script of the Autoupgrade module, as it can be seen in this [PR](https://github.com/PrestaShop/autoupgrade/pull/672/files). 
```

Reviewer 2: Here, we also have to be more precise.

"In order to have this new hook available for users upgrading from older versions of PrestaShop" would sound better.

Also, we have to be a bit more precise what it means "current version SQL script for autougrade module", maybe a suggestion how to check which version will be next, or just additional text saying something like:

"If you are unsure to which file you should put the code inside `autoupgrade` module, feel free to ping "here's committers tag on github""

Also, consider mentioning this method of adding the new hook:
https://github.com/PrestaShop/autoupgrade/pull/577

# How to add a new Hook
Adding a new hook is a process that can be done in only three steps.

## 1) Add a call to Hook::exec()

> As you can see on this [PR](https://github.com/PrestaShop/PrestaShop/pull/34431/files), adding a simple Hook::exec() statement is enough to add a new hook. In order for it to be complete, there is a couple more steps to do.

Reviewer 2: ```suggestion
As visible in [this PR](https://github.com/PrestaShop/PrestaShop/pull/34431/files), the first step is to add a `Hook::exec()` in the code where you want to execute it. This will execute the code from the modules hooked into your `Hook`.
```

## 2) Modify the hook.xml file

> In order to properly register the hook you need to add it to the hook.xml file as seen in this [PR](https://github.com/PrestaShop/PrestaShop/pull/34431/files)

Reviewer 2: ```suggestion
The second step is correctly adding the `Hook` during installation. You need to add it to the `hook.xml` file, as seen in this [PR](https://github.com/PrestaShop/PrestaShop/pull/34431/files). This will ensure that the `Hook` is added to the database, which enables users to sort modules hooked into it. It won't be possible without the hook in the database.
```

## 3) Add the new hook in the Autoupgrade fixture

> Finally, in order to have this new hook available for users upgrading from older versions of PrestaShop, you need to add it to the current version SQL script of the Autoupgrade module, as it can be seen in this [PR](https://github.com/PrestaShop/autoupgrade/pull/672/files).

Reviewer 2: ```suggestion
The last step is to make this new hook available for users upgrading from older versions of PrestaShop. You need to add it to the current version of the SQL script of the Autoupgrade module, as can be seen in this [PR](https://github.com/PrestaShop/autoupgrade/pull/672/files).
```

> If you are unsure to which file you should put the code inside autoupgrade module, feel free to ping `@PrestaShop/Committers` 

Reviewer 2: ```suggestion
If you are unsure which file to put the code inside the [autoupgrade module](https://github.com/PrestaShop/autoupgrade), feel free to ping `@PrestaShop/Committers`.
```

> There is also a [helper](https://github.com/PrestaShop/autoupgrade/pull/577/files) available to add a new hook.

Reviewer 2: ```suggestion
You can also use a [helper](https://github.com/PrestaShop/autoupgrade/pull/577/files) available to add a new hook during the upgrade process for a given version.

Following these three steps will ensure that the hook is correctly:

- added to the database and available to sort modules in it by position
- executed in the code
- added for users upgrading from the old versions

```
